src/PKSim.Core/Snapshots/Mappers/SnapshotMapperBase.cs

```diff
@@ -51,13 +61,36 @@ public static Task FromException(Exception exc)
          tcs.SetException(exc);
          return tcs.Task;
       }
+
+      /// <summary>
+      ///    Maps a list of models to the corresponding snapshot arrays. If the list if null or empty, null will be returned
+      /// </summary>
+      public virtual Task<TSnapshot[]> MapToSnapshots(IEnumerable<TModel> models) => MapToSnapshots(models, MapToSnapshot);
```

Comment:
Those new methods will remove the use of Task.WhenAll everywhere. The mapper takes care of it